### PR TITLE
Fix RF-DETR argpartition out-of-bounds when num_select exceeds candidates

### LINKED ISSUE #1449: RF-DETR postprocessing fails when num_select >= available candidates
### Description

RF-DETR inference can fail when `num_select` is greater than or equal to the number of available candidates in the model output.

In [anylabeling/services/auto_labeling/rfdetr.py](anylabeling/services/auto_labeling/rfdetr.py), the postprocess logic does:

```python
topk_indexes = np.argpartition(
    -prob_reshaped, self.num_select, axis=1
)[:, : self.num_select]
```

This is problematic because `np.argpartition` requires `kth` to be strictly smaller than the number of flattened logits. If `num_select` is equal to or greater than the candidate count, NumPy raises an out-of-bounds error.

### Steps to reproduce

1. Configure RF-DETR with a `num_select` value that is equal to or greater than the number of available candidates
2. Run inference with a model output whose flattened logits count is smaller than or equal to that value
3. The model crashes during postprocessing

### Expected behavior

Inference should continue normally and use only the available candidates instead of failing during top-k selection.

### Actual behavior

The process raises a NumPy `ValueError` during `argpartition`, causing inference to fail.

### PR DESCRIPTION
**Pull request by MNakahara-ishidatec**
## Summary

Fix RF-DETR postprocessing when `num_select` is equal to or greater than the number of available candidates.

## Root cause

`self.num_select` was passed directly as the `kth` argument to `np.argpartition`.

`kth` is a zero-based index, so it must be strictly smaller than the number of flattened logits. When `num_select` reaches or exceeds the candidate count, `kth` becomes out of bounds and NumPy raises a `ValueError`.

## Changes

- Clamp `num_select` to the actual number of available candidates
- Use the valid zero-based index `kth = max(0, min(num_select, candidate_count) - 1)`
- Add a regression test covering the boundary and overflow conditions

## Validation

Ran:

```text
.\.venv\Scripts\python.exe -m pytest tests/test_models/test_dfine_seg.py -q
```

result:
```text
3 passed in 1.92s
```

fixes: #1449

- [x] I have read and agree to the CLA.

### CODE REVIEW

**Comment by CVHub520** [comment]
Thanks for the fix! Merged into main with follow-up changes preserving zero-selection behavior and expanding regression coverage.
